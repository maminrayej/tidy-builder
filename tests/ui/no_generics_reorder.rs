#[derive(tidy_builder::Builder)]
struct MyStruct {
    req1: String,
    req2: String,
    opt1: Option<String>,
}

fn main() {
    let my_struct = MyStruct::builder()
        .req1("req1".to_string())
        .build();

    assert_eq!(my_struct.req1, "req1".to_string());
    assert_eq!(my_struct.req2, "req2".to_string());
    assert_eq!(my_struct.opt1, Some("opt1".to_string()));
}
