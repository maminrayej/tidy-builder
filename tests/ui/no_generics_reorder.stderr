error[E0599]: no method named `build` found for struct `MyStructBuilder<true, false>` in the current scope
  --> tests/ui/no_generics_reorder.rs:11:10
   |
1  | #[derive(tidy_builder::Builder)]
   |          --------------------- method `build` not found for this
...
11 |         .build();
   |          ^^^^^ method not found in `MyStructBuilder<true, false>`
   |
   = note: the method was found for
           - `MyStructBuilder<true, true>`
